firedrake/tsfc_interface.py

```diff
@@ -181,6 +182,8 @@ def compile_form(form, name, parameters=None, inverse=False, split=True, interfa
         parameters = default_parameters["form_compiler"].copy()
         parameters.update(_)
 
+    parameters['scalar_type'] = utils.ScalarType_c
```

Comment:
Would it have made sense to set this in the `default_parameters["form_compiler"]` dict, rather than having to remember where-ever we use the parameters.

Reply:
I am making an attempt to do this.